### ISSUE TITLE
Enable CSS and JavaScript bundling feature

### ISSUE THREAD

**Issue by WebDevJL** [Puzzlout/TrekToursFlyer#12]
See the [Section CSS and JS bundling](https://docs.google.com/document/d/1n48pjUYmQUl6SUm5HrUOV3PSO9fSiWqQL8aZEdkVVTs/edit#heading=h.qowucm2hoqjc).

We will use the Assetic Bundle from Symphony. Please implement it in the application in order to bundle into one HTTP request all the css and javascript.

**_Note:**_ do split the issue into "Enable CSS bundling feature" and "Enable JavaScript bundling feature" if it takes longer than the 10h max per task allowance and link the new issue(s) to this one.

**Comment by damirius** [comment]
@WebDevJL 

You can find the code on feature_appoptimisations branch.
I've set some css files through assetic filter.
Assetic works like this, on dev it will still serve each files separately. On production it will serve one file (depending where we used it).
After fetching the branch run `composer update` and `php bin/console assetic:dump --env=prod --no-debug` which will create combined files for css. For now I just combined one in layout.html.twig, since we don't have much.
Question is if we are moving to bower or not, because this two issues are closely related.

**Comment by WebDevJL** [comment]
@damirius Bower is like Composer but for CSS and JS packages, right? It can be good to keep the third party packages up-to-date. You tackle the task of enabling Bower for the project if you want.

For now, I will try the bundling stuff you did.

**Comment by WebDevJL** [comment]
@damirius a few question for you:
- how do you decide in which order you combine each css file? In fact, we may need to include some css before some other.
- does it work for js files as well? That'd be nice!

Thanks!

**Comment by damirius** [comment]
@WebDevJL 

I believe you can add them up by listing them one by one like this example:

```
{% javascripts
    '@AppBundle/Resources/public/js/*'
    '@AcmeBarBundle/Resources/public/js/form.js'
    '@AcmeBarBundle/Resources/public/js/calendar.js' %}
    <script src="{{ asset_url }}"></script>
{% endjavascripts %}
```

You can still use `*` wildcard to import everything else also.
As you can see from example it works with javascript, just use `{% javascripts %}` tag instead of `{% stylesheets %}`

**Comment by WebDevJL** [comment]
@damirius Ok. We'll probably activate that if the js file list gets long.

For now, I will try to get the css bundling to work.

**Comment by WebDevJL** [comment]
@damirius Ok for me. Are you done with the task or are you waiting on me for a detail on something?

**Comment by damirius** [comment]
@WebDevJL

Well we can keep it open if you wish, but I'll move onto another one to tackle that bower issue.
Edit: Is it ok to move that bower issue to this milestone so we can work on it on the same branch?

**Comment by WebDevJL** [comment]
@damirius Yes.

**Comment by WebDevJL** [comment]
@gsouvik 

You must install bower from now on as I merged its addition to the project just now. Here are the steps:
- download the nodejs pkg (https://nodejs.org/en/download/)
- install nodejs
- in terminal, 
  - run `sudo su -` 
  - run `npm install -g bower`
  - run `bower -v` to check that bower works.
  - run `cd root/path/to/project`
  - run `bower install`
  - and finally, run `sh sh/refresh.sh dev`

Let me know if you have any issues.
